### PR TITLE
perf(dsv41): fuse the FP8 index-query quantization into one kernel

### PR DESCRIPTION
**Pull request by wangbo981016**
## Motivation

Preparing index queries for the FP8 logits kernel — quantizing them to E4M3 and folding the per-(token, head) scale into the score weights, since DeepGEMM's logits kernels apply no query scale of their own — was four lines of eager tensor ops:

```python
values = index_q.float()
scale = values.abs().amax(-1, keepdim=True).clamp_min(1.0e-6) / 448.0
quantized = (values / scale).clamp(-448.0, 448.0).to(torch.float8_e4m3fn)
folded = (weights.float() * scale.squeeze(-1)).contiguous()
```

That expands to eleven kernels: a float cast, `abs`, an `amax` reduction, `clamp_min`, two divides, `clamp`, the E4M3 cast, a contiguous copy, the weight multiply, and another copy. They all run over `[tokens, heads, 128]`, which for a decode step is 32768 elements, so each one is pure launch latency.

Timed on the indexer stream of a captured decode, that prologue is 19 us against 16 us for the paged MQA logits kernel it feeds. Query preparation costs more than the scoring it prepares for.

## Changes

One Triton kernel replaces the chain. A program per `(token, head)` reads its 128-element row, takes the absolute maximum, writes the E4M3 payload, and folds the scale straight into the weight it emits alongside. The separate scale tensor and both contiguous copies disappear with it, and the caller drops from two statements to one.

The last-dimension stride is passed explicitly, so a head-major query view is read in place rather than copied first.

`index_q_quantize` was not reusable here: it simulates the cache codec by packing and unpacking back to BF16 for the portable scorer, so it yields neither an FP8 payload nor the scale this path has to fold.

## Tests

`tokenspeed-kernel/test/ops/test_attention_dsv41.py`: 103 passed, 1 skipped.

Adds `test_index_query_quantization_matches_the_reference_codec`, which asserts the fused kernel reproduces the eager chain **bit for bit** — the E4M3 payload compared as raw bytes and the folded weight at `rtol=0, atol=0`. The payload decides which rows a pass goes on to score, so an off-by-one-ulp rounding difference would silently change selection. Parametrized over a contiguous and a strided (head-major) query so the in-place read is covered.

### CODE REVIEW

**Comment by wangbo981016** [comment]
@codex review

File: tokenspeed-kernel/python/tokenspeed_kernel/ops/attention/dsv41/triton.py [hunk 1]
```diff
@@ -1678,6 +1678,68 @@ def _pack_index_queries(X, V, S, X0, X1):
...
+    F1,
+    D: tl.constexpr,
+):
+    row = tl.program_id(0).to(tl.int64)
+    head = tl.program_id(1).to(tl.int64)
+    d = tl.arange(0, D)
+    value = tl.load(Q + row * Q0 + head * Q1 + d * Q2).to(tl.float32)
+    scale = tl.maximum(tl.max(tl.abs(value), axis=0), 1.0e-6) / 448.0
+    quantized = tl.minimum(tl.maximum(value / scale, -448.0), 448.0)
+    tl.store(QOut + row * O0 + head * O1 + d, quantized.to(tl.float8e4nv))
```
[the remark this answers is not in the document]
Comment (wangbo981016) [reply tokenspeed-kernel/python/tokenspeed_kernel/ops/attention/dsv41/triton.py]: Fixed in 5ccc394b7, with one deliberate deviation.

The payload division now uses `tl.div_rn` and the launch sets `enable_fp_fusion=False`, matching the other exact codecs in this module.

The scale keeps the multiply by `1 / 448`. PyTorch divides a tensor by a Python scalar on its CPU-scalar fast path as `a * (1 / b)`, so the reference `amax.clamp_min(1e-6) / 448.0` is a reciprocal multiply, not a division. Switching that step to `tl.div_rn` breaks bit-exactness against the reference (26 of 20480 payload bytes in the test differ by one E4M3 code); with the reciprocal multiply the whole file passes bit for bit. This also matches the `index_v4` key codec, which computes `raw_scale = amax * (1.0 / 448.0)`.

**Comment by wangbo981016** [comment]
@codex review
